lib/util/state.js

```diff
@@ -106,75 +106,116 @@ export function getActiveItineraries (otpState) {
           if (direction === 'ASC') return a.duration - b.duration
           else return b.duration - a.duration
         case 'COST':
-          const aTotal = getTotalFare(a)
-          const bTotal = getTotalFare(b)
+          const aTotal = getTotalFare(a, config)
+          const bTotal = getTotalFare(b, config)
           if (direction === 'ASC') return aTotal - bTotal
           else return bTotal - aTotal
         default:
           if (type !== 'BEST') console.warn(`Sort (${type}) not supported. Defaulting to BEST.`)
           // FIXME: Fully implement default sort algorithm.
-          const aCost = calculateItineraryCost(a)
-          const bCost = calculateItineraryCost(b)
+          const aCost = calculateItineraryCost(a, config)
+          const bCost = calculateItineraryCost(b, config)
           if (direction === 'ASC') return aCost - bCost
           else return bCost - aCost
       }
     })
 }
 
+// Default constants for calculating itinerary "cost", i.e., how preferential a
+// particular itinerary is based on factors like wait time, total fare, drive
+// time, etc.
+const FARE_FACTOR = 0.5
+const DURATION_FACTOR = 0.25
+const WALK_RELUCTANCE = 0.1
+const WAIT_RELUCTANCE = 0.1
+const DRIVE_RELUCTANCE = 2
+const TRANSFER_RELUCTANCE = 0.9
+
 /**
  * This calculates the "cost" (not the monetary cost, but the cost according to
  * multiple factors like duration, total fare, and walking distance) for a
  * particular itinerary, for use in sorting itineraries.
  * FIXME: Do major testing to get this right.
  */
-function calculateItineraryCost (itinerary) {
-  // FIXME: Add drive reluctance factor?
-  // FIXME: Make these configurable.
-  const FARE_COST_FACTOR = 0.5
-  const DURATION_COST_FACTOR = 0.25
-  const WALK_COST_FACTOR = 0.1
-  const WAIT_COST_FACTOR = 0.1
-  const TRANSFER_COST_FACTOR = 0.9
-  return getTotalFare(itinerary) * FARE_COST_FACTOR +
-    itinerary.duration * DURATION_COST_FACTOR +
-    itinerary.walkDistance * WALK_COST_FACTOR +
-    itinerary.waitingTime * WAIT_COST_FACTOR +
-    itinerary.transfers * TRANSFER_COST_FACTOR
+function calculateItineraryCost (itinerary, config = {}) {
+  const weights = {
```

Comment:
Can you define a `defaultWeights` variable instead of writing out each variable individually?